packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -166,6 +167,16 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     return workInProgress.child;
   }
 
+  function updateUseRef(current, workInProgress) {
+    const nextChildren = workInProgress.type.renderProp(
```

Comment:
Might want to validate before calling to get a nicer error message. See a similar check before context. I think Seb told me once it doesn’t affect perf because engine has to check it’s a function regardless and will likely remember that.

Reply:
I already validate in `packages/react/src/useRef.js` when the prop is passed initially.

Reply:
The validation in `useRef.js` is just a warning though, maybe use an invariant and throw early instead?

Reply:
Hmm maybe we should change context to work the same way then.

I don’t think we should make it ever invariant during creation. Makes inlining harder and will likely cause duplicate checks.

Reply:
Is the intention to rely on the warning to explain the error that would occur here when `renderProp` isn't a function? I understand not throwing during creation, but an explicit error seems better to me.

Reply:
I think warning during creation is a good idea, but I also think there should be an explicit error with a component stack trace when `updateUseRef` inevitably throws.

Reply:
Is 166d236 what you had in mind? The component stack will only be included in DEV mode.

This feels a little weird to me, checking both places.